zebra-network/src/constants.rs

```diff
@@ -511,7 +515,8 @@ mod tests {
         );
 
         assert!(
-            MAX_ADDRS_IN_ADDRESS_BOOK < TYPICAL_MAINNET_ADDRESS_BOOK_SIZE,
+            MAX_ADDRS_IN_ADDRESS_BOOK
+                <= TYPICAL_MAINNET_ADDRESS_BOOK_SIZE.max(MINIMUM_MAINNET_ADDRESS_BOOK_SIZE),
             "the address book limit should actually be used"
```

Comment:
`MINIMUM_MAINNET_ADDRESS_BOOK_SIZE ` is the same value as `MAX_ADDRS_IN_ADDRESS_BOOK`, so the check will always be true, regardless of the values of `MAX_ADDRS_IN_MESSAGE` or `ADDR_RESPONSE_LIMIT_DENOMINATOR`.

Instead, I suggest modifying the check so it is still meaningful, and we have to ask ourselves "is the limit too big to actually be used?" next time we change the peer set size. (That's the point of these kinds of checks.)

See my suggestion above as well.
```suggestion
            MAX_ADDRS_IN_ADDRESS_BOOK <= TYPICAL_MAINNET_ADDRESS_BOOK_SIZE,
            "the address book limit should actually be used"
```